### PR TITLE
feat: support file mode on default stream mode

### PR DESCRIPTION
**Pull request by fengmk2**
using `options.fileModeMatch` to match the requests.

##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: test/stream-mode-with-filematch.test.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      headers,
+      stream: form,
+    });
+
+    assert(res.status === 200);
+    const data = JSON.parse(res.data);
+    assert.deepStrictEqual(data, { body: {} });
+  });
+
+  it('should register clean_tmpdir schedule', () => {
```
Comment (fengmk2) [inline test/stream-mode-with-filematch.test.js]: 之前少了这个测试。

File: app/schedule/clean_tmpdir.js [hunk 2]
```diff
@@ -12,8 +12,8 @@ module.exports = app => {
         type: 'worker',
         cron: app.config.multipart.cleanSchedule.cron,
         immediate: false,
-        // disable on stream mode
-        disable: app.config.multipart.mode === 'stream',
+        // disable on stream mode and not set fileModeMatch
+        disable: app.config.multipart.mode === 'stream' && !app.config.multipart.fileModeMatch,
```
Comment (fengmk2) [inline app/schedule/clean_tmpdir.js]: 测试不能少，要不然这里就很容易忽略了。

**Comment by dead-horse** [comment]
2.6.0

**Comment by atian25** [comment]
d.ts 也要加下

**Pull request merged**